### PR TITLE
Add table aws_vpc_verified_access_group closes #1468

### LINKED ISSUE #1468: Add table aws_vpc_verified_acces_group
**References**
https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeVerifiedAccessGroups.html

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select
  verified_access_group_id,
  arn,
  verified_access_instance_id,
  creation_time,
  description,
  last_updated_time
from
  aws_vpc_verified_access_group;
+--------------------------+---------------------------------------------------------------------------------+-----------------------------+---------------------------+-------------+----------------------
| verified_access_group_id | arn                                                                             | verified_access_instance_id | creation_time             | description | last_updated_time    
+--------------------------+---------------------------------------------------------------------------------+-----------------------------+---------------------------+-------------+----------------------
| vagr-0787dfc188e8ca2c2   | arn:aws:ec2:us-east-2:873402157654:verified-access-group/vagr-0787dfc188e8ca2c2 | vai-011ed46e43d2d1b1e       | 2022-12-05T18:38:35+05:30 | testing     | 2022-12-05T18:38:35+0
| vagr-018739a48d861ff14   | arn:aws:ec2:us-east-2:873402157654:verified-access-group/vagr-018739a48d861ff14 | vai-011ed46e43d2d1b1e       | 2022-12-05T14:01:01+05:30 | test        | 2022-12-05T14:01:01+0
+--------------------------+---------------------------------------------------------------------------------+-----------------------------+---------------------------+-------------+----------------------

> select
  verified_access_group_id,
  creation_time,
  description,
  last_updated_time
from
  aws_vpc_verified_access_group
where
  creation_time >= now() - interval '30' day;
+--------------------------+---------------------------+-------------+---------------------------+
| verified_access_group_id | creation_time             | description | last_updated_time         |
+--------------------------+---------------------------+-------------+---------------------------+
| vagr-0787dfc188e8ca2c2   | 2022-12-05T18:38:35+05:30 | testing     | 2022-12-05T18:38:35+05:30 |
| vagr-018739a48d861ff14   | 2022-12-05T14:01:01+05:30 | test        | 2022-12-05T14:01:01+05:30 |
+--------------------------+---------------------------+-------------+---------------------------+

> select
  g.verified_access_group_id,
  g.creation_time,
  i.creation_time as instance_create_time,
  i.verified_access_instance_id,
  jsonb_pretty(i.verified_access_trust_providers) as verified_access_trust_providers
from
  aws_vpc_verified_access_group as g,
  aws_vpc_verified_access_instance as i
where
  g.verified_access_instance_id = i.verified_access_instance_id;
+--------------------------+---------------------------+---------------------------+-----------------------------+-------------------------------------------------------------------+
| verified_access_group_id | creation_time             | instance_create_time      | verified_access_instance_id | verified_access_trust_providers                                   |
+--------------------------+---------------------------+---------------------------+-----------------------------+-------------------------------------------------------------------+
| vagr-018739a48d861ff14   | 2022-12-05T14:01:01+05:30 | 2022-12-05T14:00:40+05:30 | vai-011ed46e43d2d1b1e       | [                                                                 |
|                          |                           |                           |                             |     {                                                             |
|                          |                           |                           |                             |         "Description": "test",                                    |
|                          |                           |                           |                             |         "TrustProviderType": "user",                              |
|                          |                           |                           |                             |         "UserTrustProviderType": "iam-identity-center",           |
|                          |                           |                           |                             |         "DeviceTrustProviderType": "",                            |
|                          |                           |                           |                             |         "VerifiedAccessTrustProviderId": "vatp-0f71337c0114b9e3e" |
|                          |                           |                           |                             |     }                                                             |
|                          |                           |                           |                             | ]                                                                 |
| vagr-0787dfc188e8ca2c2   | 2022-12-05T18:38:35+05:30 | 2022-12-05T14:00:40+05:30 | vai-011ed46e43d2d1b1e       | [                                                                 |
|                          |                           |                           |                             |     {                                                             |
|                          |                           |                           |                             |         "Description": "test",                                    |
|                          |                           |                           |                             |         "TrustProviderType": "user",                              |
|                          |                           |                           |                             |         "UserTrustProviderType": "iam-identity-center",           |
|                          |                           |                           |                             |         "DeviceTrustProviderType": "",                            |
|                          |                           |                           |                             |         "VerifiedAccessTrustProviderId": "vatp-0f71337c0114b9e3e" |
|                          |                           |                           |                             |     }                                                             |
|                          |                           |                           |                             | ]                                                                 |
+--------------------------+---------------------------+---------------------------+-----------------------------+-------------------------------------------------------------------+

> select
  verified_access_group_id, 
   verified_access_instance_id,
  creation_time,
  deletion_time,
  description,
  last_updated_time
from
  aws_vpc_verified_access_group where verified_access_instance_id = 'vai-017531e6f0e305efc'
+--------------------------+-----------------------------+---------------------------+---------------+---------------+---------------------------+
| verified_access_group_id | verified_access_instance_id | creation_time             | deletion_time | description   | last_updated_time         |
+--------------------------+-----------------------------+---------------------------+---------------+---------------+---------------------------+
| vagr-066e6e9e706ee46c4   | vai-017531e6f0e305efc       | 2022-12-13T19:24:48+05:30 | <null>        | testing group | 2022-12-13T19:24:48+05:30 |
+--------------------------+-----------------------------+---------------------------+---------------+---------------+---------------------------+

```
</details>

### CODE REVIEW

**Comment by ParthaI** [comment]
Go lang lint suppose to fail until the [PR](https://github.com/turbot/steampipe-plugin-aws/pull/1482) gets merged into the main.

File: aws/table_aws_vpc_verified_acces_group.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "verified_access_group_id", Require: plugin.Optional},
+				{Name: "verified_access_instance_id", Require: plugin.Optional},
+			},
+		},
+		GetMatrixItemFunc: BuildRegionList,
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "verified_access_group_id",
+				Description: "The ID of the Verified Access group.",
```
Comment (misraved) [inline aws/table_aws_vpc_verified_acces_group.go:34]:
```suggestion
				Description: "The ID of the verified access group.",
```

I believe all the instances of the `verified access group` can be updated as per the above suggestion.

**Review by misraved — COMMENTED** [review]
@ParthaI please take a look at the suggestion. Thanks!!

The file name of the table is incorrect and should be updated to use `aws_vpc_verified_access_group` instead of `aws_vpc_verified_acces_group`.

Could you please add a few example queries wherein the `verified_access_group_id` and `verified_access_instance_id`? Curious to see the usage of these optional qual columns.

File: aws/table_aws_vpc_verified_access_group.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+			Hydrate: getVpcVerifiedAccessGroup,
+		},
+		List: &plugin.ListConfig{
+			Hydrate: listVpcVerifiedAccessGroups,
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"InvalidParameterValue"}),
+			},
+			// DescribeVerifiedAccessGroups API accept group id as input param.
+			// We are passing MaxResults value as DescribeVerifiedAccessGroups api input
+			// We can not pass both MaxResults and VerifiedAccessGroupId at a time in the same input, we have to pass either one. So verified_access_group_id can not be added as optional quals and added get config for filtering out the group by their id.
```
Comment (misraved) [inline aws/table_aws_vpc_verified_access_group.go:34]:
```suggestion
			// We cannot pass both MaxResults and VerifiedAccessGroupId at a time in the same input, we have to pass either one. So verified_access_group_id can not be added as optional quals. Added get config for filtering out the group by its id.
```

File: aws/table_aws_vpc_verified_access_group.go [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+	}
+
+	// Get call
+	op, err := svc.DescribeVerifiedAccessGroups(ctx, input)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_vpc_verified_access_group.getVpcVerifiedAccessGroup", "api_error", err)
+		return nil, err
+	}
+
+	if op.VerifiedAccessGroups != nil && len(op.VerifiedAccessGroups) > 0 {
```
Comment (misraved) [inline aws/table_aws_vpc_verified_access_group.go:201]:
```suggestion
	if op.VerifiedAccessGroups != nil || len(op.VerifiedAccessGroups) > 0 {
```

Do we need both the conditions to be true?

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!

File: aws/table_aws_vpc_verified_access_group.go (hunk 3 above)
Comment (ParthaI) [reply aws/table_aws_vpc_verified_access_group.go:201]: Yes, we need both condition to be true, if we will put `||` then `len()` function will throw when `op.VerifiedAccessGroups` value will be nil. `len()` only works on string and array not on `nil`

**Pull request merged**